mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -139,6 +142,27 @@ public boolean onPreferenceClick(Preference preference) {
                     return true;
                 }
             });
+
+            clearCachePreference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+                @Override
+                public boolean onPreferenceClick(Preference preference) {
+                    if (getActivity() == null) {
+                        return false;
+                    }
+                    // Get launch intent for application
+                    Intent restartIntent = getActivity().getBaseContext().getPackageManager()
+                            .getLaunchIntentForPackage(getActivity().getBaseContext().getPackageName());
+                    restartIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    // Finish current activity
+                    getActivity().finish();
```

Comment:
Wouldn't you rather want to set an activity result here, start the settings with startActivityForResult and restart the main activity from its onActivityResult? And there also is Activity::recreate, I don't think that complicated stuff of really needed.

Reply:
When I think about that, I'm not really sure if all this is really needed. Isn't the main activity restarted when I leave the settings menu? In this case, the cleared image cache should already be used and filled with new values? So probably just a message like "Image cache cleared" should be fine? What do you think?